systems/src/apu/mod.rs

```diff
@@ -0,0 +1,1396 @@
+        }
+
+        #[test]
+        fn start_contactor_is_not_energised_when_shutting_down() {
+            let mut tester = tester_with()
+                .running_apu()
+                .then_continue_with()
+                .master_off();
+
+            loop {
+                tester = tester.run(Duration::from_millis(50));
+                assert_eq!(tester.start_contactor_energized(), false);
+
+                if tester.get_n().get::<percent>() == 0. {
+                    break;
+                }
+            }
+        }
+
+        #[test]
+        fn start_contactor_is_not_energised_when_running() {
+            let tester = tester_with().running_apu().run(Duration::from_secs(1_000));
+            assert_eq!(tester.start_contactor_energized(), false);
+        }
+
+        #[test]
+        fn start_contactor_is_not_energised_when_shutting_down_with_master_on_and_start_on() {
+            let mut tester = tester_with()
+                .running_apu_without_bleed_air()
+                .and()
+                .master_off()
+                .run(Duration::from_secs(1));
+
+            while tester.apu_is_available() {
+                tester = tester.run(Duration::from_secs(1));
+            }
+
+            tester = tester.master_on().and().start_on();
+
+            let mut n = 100.;
+
+            while n > 0. {
+                // Assert before running, because otherwise we capture the Starting state which begins when at n = 0
+                // with the master and start switches on.
+                assert_eq!(tester.start_contactor_energized(), false);
+
+                tester = tester.run(Duration::from_secs(1));
+                n = tester.get_n().get::<percent>();
+            }
+        }
+
+        #[test]
+        fn available_when_n_above_99_5_percent() {
+            let mut tester = tester_with().starting_apu();
+
+            loop {
+                tester = tester.run(Duration::from_millis(50));
+                let n = tester.get_n().get::<percent>();
+                assert!((n > 99.5 && tester.apu_is_available()) || !tester.apu_is_available());
+
+                if n == 100. {
+                    break;
+                }
+            }
+        }
+
+        #[test]
+        fn available_when_n_above_95_percent_for_2_seconds() {
+            let mut tester = tester_with()
+                .starting_apu()
+                .and()
+                .turbine_infinitely_running_at(Ratio::new::<percent>(96.))
+                .run(Duration::from_millis(1999));
+
+            assert!(!tester.apu_is_available());
+
+            tester = tester.run(Duration::from_millis(1));
+            assert!(tester.apu_is_available());
+        }
+
+        #[test]
+        #[timeout(500)]
+        fn without_fuel_apu_starts_until_approximately_n_3_percent_and_then_shuts_down_with_fault()
+        {
+            let mut tester = tester_with()
+                .no_fuel_available()
+                .run(Duration::from_secs(1_000))
```

Comment:
I'm curious why this is needed?

Edit: Answered below in one of your comments. Maybe add a comment here as well?

Reply:
I agree there is some funkiness in these tests. Adapting them to be clearer.